yet-another-docker-plugin/src/main/resources/com/github/kostyasha/yad/launcher/DockerComputerJNLPLauncher/init.sh

```diff
@@ -59,10 +59,12 @@ if [ ! -z "$COMPUTER_SECRET" ]; then
  RUN_CMD+=" -secret $COMPUTER_SECRET"
 fi
 
-if [[ -x $(command -v gosu) ]]; then
-    RUN_CMD="gosu $JENKINS_USER $RUN_CMD"
-elif [ ! -z "$JENKINS_USER" ] && [ x"$JENKINS_USER" != "xroot" ]; then
-    RUN_CMD="su - $JENKINS_USER -c '$RUN_CMD'"
+if [ "$(id -nu)" != "$JENKINS_USER" ]; then
+    if [ -x "$(command -v gosu)" ]; then
+        RUN_CMD="gosu $JENKINS_USER $RUN_CMD"
+    else
+        RUN_CMD="su - $JENKINS_USER -c '$RUN_CMD'"
```

Comment:
My main consideration was not using it at all when it's not necessary (a scenario where it _did_ fail).  I did not investigate other cases where `su` may fail.